### ISSUE TITLE
False positive on not-callable when inheriting from `typing.IO`

### ISSUE THREAD

**Issue by Redoubts** [pylint-dev/pylint#10042]
### Bug description

If I write a class which inherits the `typing.IO` abc, pylint complains that class instantiation is not callable when I provide a typevar as well (eg `X[T]()`, something like `list[str]()`). I think I should be allowed to do this, especially in this case where I want to be explicit about the TypeVar's real type.

(lazy) Example. The error is at `Z[str]()`, even though `X[str]()` and `Y[str]()` are fine.
```
import typing


class X(typing.Generic[typing.AnyStr]): ...


class Y(X[typing.AnyStr]): ...


class Z(typing.IO[typing.AnyStr]):
    """pretend full implementation here..."""


X[str]()
Y[str]()
Z[str]()  # not-callable
```

### Command used

pylint x.py

### Pylint output

************* Module x
x.py:16:0: E1102: Z[str] is not callable (not-callable)

------------------------------------------------------------------
Your code has been rated at 4.44/10 (previous run: 0.00/10, +4.44)

### Expected behavior

10/10

### Pylint version

pylint 3.3.1
astroid 3.3.5
Python 3.11.10

### OS / Environment

Debian 11
